### PR TITLE
Documentation updates around authentication & authorisation

### PR DESCRIPTION
**Pull request by SarahFrench**
# Description

This PR updates the main documentation page for the provider, that includes auth info:

- Added more details in the index page about the default `oath_scopes` value, and how `oath_scopes` should be used to explicitly set scoped in a provider configuration even though it's not a required field.
- Added more examples of provider configuration
- Moved an existing guide doc into a new folder to allow it to be shown in a Guides section in the Registry UI ([see explanation here in Registry docs](https://www.terraform.io/registry/providers/docs#guides-subcategories))

### CODE REVIEW

**Comment by SarahFrench** [comment]
Keeping this as a draft until we've investigated generating separate docs pages

**Comment by SarahFrench** [comment]
I found that the pre-existing guide doc wasn't appearing on the Registry because it needs to be in a `guides` folder - [see docs here](https://www.terraform.io/registry/providers/docs#guides-subcategories). In this PR I've moved the existing guide, and will hold off on making auth guides separate to the main index page

File: examples/provider/provider-service-account-with-admin-roles.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Auth method: Admin roles applied directly to a service account
```
Comment (megan07) [inline examples/provider/provider-service-account-with-admin-roles.tf:1]: Have you tested this with any resources outside of groups etc? I believe this is only applicable to groups, but I may be wrong. If it is, I think we should make a note that Service Accounts are only able to be granted Groups Admin, in which case they can only manage the groups scope

Comment (SarahFrench) [reply examples/provider/provider-service-account-with-admin-roles.tf:1]: I've had a play around with service accounts given admin roles directly and users can be managed as well as groups -I've made a new acceptance test that demonstrates this but it requires some changes to the acceptance test infra: https://github.com/hashicorp/terraform-provider-googleworkspace/pull/325

I've tested some other things and seen they don't work without impersonation - I'll update the docs with this information!

File: docs/index.md [hunk 2]
```diff
@@ -84,8 +84,13 @@ Only users with access to the Admin APIs can access the Admin SDK Directory API,
 must be set in the environment variable `GOOGLEWORKSPACE_IMPERSONATED_USER_EMAIL` or in the `impersonated_user_email` attribute in the provider. Additionally, the user must have logged in at least once and accepted the Google Workspace Terms of Service.
 
 ### Using Specific Administrator Roles
+
 You do not need to set up domain-wide delegation if you are granting more specific administrator roles to the service account. If the Terraform pipeline execution environment provides an appropriate token as Application Default Credentials (ADC), you can use the provider without any further setup.
 
+To do this it's recommended that you create a custom admin role(s) with the Admin API privileges you need, as [pre-built administrator roles](https://support.google.com/a/answer/2405986) might not cover your use case.
+
+~> Some resources controlled via this provider can only be managed by a Super Admin, so user impersonation is necessary in those cases. For instance, [Admin Roles can only managed via the console or API by a Super Admin](https://support.google.com/a/answer/2406043?hl=en). Also, access to some API endpoints may not be possible to grant as privileges in custom roles.
```
Comment (SarahFrench) [inline docs/index.md:92]: `~>` Results in formatting like this:

Screenshot 2022-06-08 at 19 43 40

File: examples/provider/provider-service-account-with-admin-roles.tf (hunk 1 above)
Comment (SarahFrench) [reply examples/provider/provider-service-account-with-admin-roles.tf:1]: Addressed this in 63173b4 - I think the limitations of using custom admin roles are:
- some resources _must_ be managed by super admins, therefore via user impersonation
- some API endpoints may not be able to be granted via an admin role

File: docs/index.md (hunk 2 above)
Comment (megan07) [inline docs/index.md:92]:
```suggestion
~> Some resources controlled via this provider can only be managed by a Super Admin, so user impersonation is necessary in those cases. For instance, [Admin Roles can only be managed via the console or API by a Super Admin](https://support.google.com/a/answer/2406043?hl=en). Also, access to some API endpoints may not be possible to grant as privileges in custom roles. 
```

**Review by megan07 — APPROVED** [review]
I think this looks good!

File: docs/index.md (hunk 2 above)
Comment (SarahFrench) [reply docs/index.md]: Life saver, thanks!

**Pull request merged**